daml-lf/transaction/src/main/protobuf/com/daml/lf/transaction.proto

```diff
@@ -127,3 +127,19 @@ message NodeRollback { // *since version 14*
 }
 
 // architecture-handbook-entry-end: Nodes
+
+message Versioned {
+  string version = 1;
+  bytes payload = 2;
+}
+
+message FatContractInstance {
+  bytes contract_id = 1;
+  com.daml.lf.value.Identifier template_id = 2;
+  bytes create_arg = 3;
+  KeyWithMaintainers contract_key_with_maintainers = 4;
+  repeated string non_maintainer_signatories = 5;
+  repeated string non_signatory_stakeholders = 6;
+  int64 created_at = 7;
+  bytes canton_data = 8; // this is used by canton to store the contractId salt.
```

Comment:
```suggestion
  // Field for Canton to store extra data; e.g., the contractId salt.
  bytes canton_data = 8;
```

either add the comment and thus leave it free to Canton to store the extra data.

Alternatively consider adding a field for the Salt directly, and thus make it such that within the whole participant node all data is accessible.

Reply:
That said: we probably don't want to do that change away from `canton_data` right now, as it likely changes the flow of implementation.

Reply:
done